app/apps/client/admin/helpers.js

```diff
@@ -0,0 +1,349 @@
+		html: false,
+	}, (confirmed) => {
+		if (!confirmed) {
+			return;
+		}
+		callback();
+	});
+};
+
+const promptAppUninstall = (isSubscribed, callback) => {
+	modal.open({
+		text: isSubscribed
+			? t('Apps_Marketplace_Uninstall_Subscribed_App_Prompt')
+			: t('Apps_Marketplace_Uninstall_App_Prompt'),
+		type: 'warning',
+		showCancelButton: true,
+		confirmButtonColor: '#DD6B55',
+		confirmButtonText: t('Yes'),
+		cancelButtonText: t('No'),
+		closeOnConfirm: true,
+		html: false,
+	}, (confirmed) => {
+		if (!confirmed) {
+			return;
+		}
+		callback();
+	});
+};
+
+export const triggerAppPopoverMenu = (app, currentTarget, instance) => {
+	if (!app) {
+		return;
+	}
+
+	const canAppBeSubscribed = app.purchaseType === 'subscription';
+	const isSubscribed = app.subscriptionInfo && ['active', 'trialing'].includes(app.subscriptionInfo.status);
+	const isAppEnabled = appEnabledStatuses.includes(app.status);
+
+	const handleSubscription = () => promptModifySubscription(app, async () => {
+		try {
+			await Apps.syncApp(app.id);
+		} catch (error) {
+			handleAPIError(error);
+		}
+	});
+
+	const handleViewLogs = () => {
+		FlowRouter.go('app-logs', { appId: app.id }, { version: app.version });
+	};
+
+	const handleDisable = () => promptAppDeactivation(async () => {
+		try {
+			const effectiveStatus = await Apps.disableApp(app.id);
+			warnStatusChange(app.name, effectiveStatus);
+		} catch (error) {
+			handleAPIError(error);
+		}
+	});
+
+	const handleEnable = async () => {
+		try {
+			const effectiveStatus = await Apps.enableApp(app.id);
+			warnStatusChange(app.name, effectiveStatus);
+		} catch (error) {
+			handleAPIError(error);
+		}
+	};
+
+	const handleUninstall = () => promptAppUninstall(isSubscribed, async () => {
+		try {
+			await Apps.uninstallApp(app.id);
+		} catch (error) {
+			handleAPIError(error);
+		}
+	});
+
+	popover.open({
+		currentTarget,
+		instance,
+		columns: [{
+			groups: [
+				{
+					items: [
+						...canAppBeSubscribed ? [{
+							icon: 'card',
+							name: t('Subscription'),
+							action: handleSubscription,
+						}] : [],
+						{
+							icon: 'list-alt',
+							name: t('View_Logs'),
+							action: handleViewLogs,
+						},
+					],
+				},
+				{
+					items: [
+						isAppEnabled
+							? {
+								icon: 'ban',
+								name: t('Disable'),
+								modifier: 'alert',
+								action: handleDisable,
+							}
+							: {
+								icon: 'check',
+								name: t('Enable'),
+								action: handleEnable,
+							},
+						{
+							icon: 'trash',
+							name: t('Uninstall'),
+							modifier: 'alert',
+							action: handleUninstall,
+						},
+					],
+				},
+			],
+		}],
+	});
+};
+
+export const appButtonProps = ({
+	installed,
+	version,
+	marketplaceVersion,
+	isPurchased,
+	price,
+	purchaseType,
+	subscriptionInfo,
+}) => {
+	const canUpdate = installed
+		&& version && marketplaceVersion
+		&& semver.lt(version, marketplaceVersion)
+		&& (isPurchased || price <= 0);
```

Comment:
price here shouldn't be relevant.  isPurchased should be the real key now

Reply:
```suggestion
		&& isPurchased;
```